integration-tests/eventbridge/abandon/abandon-publish.test.ts

```diff
@@ -0,0 +1,147 @@
+  deleteQueue,
+  getQueueMessages,
+  setUpQueueAndAttachToRule,
+  targetId,
+} from "../../resources/queue-helper";
+import { executeStepFunction } from "../../resources/stepfunction-helper";
+import { removeTargetFromRule } from "../../resources/event-bridge-helper";
+import { pause, retry } from "../../resources/util";
+
+describe("Abandon Step Function", () => {
+  jest.setTimeout(200_000);
+
+  const input = {
+    sessionId: "abandon-test-publish",
+    "txma-audit-encoded": "test encoded header",
+  };
+
+  let output: Partial<{
+    CommonStackName: string;
+    AbandonStateMachineArn: string;
+    AuditEventAbandonedRule: string;
+    AuditEventAbandonedRuleArn: string;
+    TxMaAuditEventRule: string;
+    TxMaAuditEventRuleArn: string;
+  }>;
+
+  let sessionTableName: string;
+  let checkHmrcEventBus: string;
+  let auditEventAbandonedRule: string;
+  let txMaAuditEventRuleName: string;
+  let abandonedEventTestQueue: CreateQueueCommandOutput;
+  let txMaAuditEventTestQueue: CreateQueueCommandOutput;
+
+  beforeEach(async () => {
+    output = await stackOutputs(process.env.STACK_NAME);
+    sessionTableName = `session-${output.CommonStackName}`;
+
+    [checkHmrcEventBus, auditEventAbandonedRule] = (
+      output?.AuditEventAbandonedRule as string
+    ).split("|");
+    [checkHmrcEventBus, txMaAuditEventRuleName] = (
+      output.TxMaAuditEventRule as string
+    ).split("|");
+
+    await populateTable(sessionTableName, {
+      sessionId: input.sessionId,
+      expiryDate: 9999999999,
+      clientId: "exampleClientId",
+      authorizationCode: "9999999999",
+      authorizationCodeExpiryDate: "9999999999",
+      clientSessionId: "252561a2-c6ef-47e7-87ab-93891a2a6a41",
+      persistentSessionId: "156714ef-f9df-48c2-ada8-540e7bce44f7",
+      clientIpAddress: "00.100.8.20",
+      subject: "test",
+    });
+
+    abandonedEventTestQueue = await setUpQueueAndAttachToRule(
+      output.AuditEventAbandonedRuleArn as string,
+      auditEventAbandonedRule,
+      checkHmrcEventBus
+    );
+    txMaAuditEventTestQueue = await setUpQueueAndAttachToRule(
+      output.TxMaAuditEventRuleArn as string,
+      txMaAuditEventRuleName,
+      checkHmrcEventBus
+    );
+  });
+  afterEach(async () => {
+    await retry(async () => {
+      await Promise.all([
+        clearItems(sessionTableName, {
+          sessionId: input.sessionId,
+        }),
+        removeTargetFromRule(
+          targetId,
+          checkHmrcEventBus,
+          txMaAuditEventRuleName
+        ),
+        removeTargetFromRule(
+          targetId,
+          checkHmrcEventBus,
+          auditEventAbandonedRule
+        ),
+      ]);
+    });
+    await retry(async () => {
+      await Promise.all([
+        deleteQueue(abandonedEventTestQueue.QueueUrl),
+        deleteQueue(txMaAuditEventTestQueue.QueueUrl),
+      ]);
+      await pause(60);
+    });
+  });
+  it("should publish ABANDONED event to CheckHmrc EventBridge Bus successfully", async () => {
```

Comment:
Looks like we're testing the messages from the `abandonedEventTestQueue` though?

Reply:
The logic for the test doesn't look inherently wrong to me. In the sense, we run a state machine and assert that the ABANDON event has actually reached a Queue. 

```java
abandonedEventTestQueue = await setUpQueueAndAttachToRule(
      output.AuditEventAbandonedRuleArn as string,
      auditEventAbandonedRule,
      checkHmrcEventBus
    );
```
It looks like it's creating a Queue with a rule attached. The rule that been set is specific for receiving only abandoned events (which already exists so it's reused). Given that, when we run the state machine, this new Queue should receive that Audit event.

Reply:
I renamed the test, let me know what you think